ui/src/components/Pagination.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { FC } from "react";
+import { PaginatedResponse } from "types/api";
+import { Button } from "@canonical/react-components";
+import { usePagination } from "util/usePagination";
+
+interface Props {
+  response?: PaginatedResponse<unknown[]>;
+}
+
+const Pagination: FC<Props> = ({ response }) => {
+  const { page, setPage } = usePagination();
+  const showFirstLink = page !== "";
+  const isMissingNext = !response || !response._meta.next;
+  const isEmptyPage = response?.data.length === 0;
+
+  if (!showFirstLink && (isMissingNext || isEmptyPage)) {
+    return null;
+  }
+
+  const first = response?._meta.first ?? "";
+  const last = response?._meta.last ?? "";
+  const next = response?._meta.next ?? "";
+  const prev = response?._meta.prev ?? "";
+
+  const leftIcon = (
+    <i className="p-icon--chevron-down" style={{ rotate: "90deg" }} />
+  );
+  const rightIcon = (
+    <i className="p-icon--chevron-down" style={{ rotate: "270deg" }} />
+  );
```

Comment:
nitpick(non-blocking): these two would probably benefit from being defined outside of the component

Reply:
Extracted those two.